### PR TITLE
Feat/UTKG -- Tests & Docs

### PR DESCRIPTION
**Pull request by GilTeraSky**
1. **What this PR does / why we need it**:
    This PR includes the tests & docs for the UTKG resource.
2. **Which issue(s) this PR fixes**
3. **Additional information**
4. **Special notes for your reviewer**

### CODE REVIEW

File: docs/guides/tanzu-mission-control_tanzu_kubernetes_cluster.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+Title: "Provisioning of a Tanzu Kubernetes Cluster"
+Description: |-
+    An example of provisioning Tanzu Kubernetes Grid Service (UTKG)
+---
+
+# Tanzu Kubernetes Cluster (UTKG)
```
Comment (willisren) [inline docs/guides/tanzu-mission-control_tanzu_kubernetes_cluster.md:7]: is it a good idea to use `UTKG` in the public doc? as it is internal term, you can think about using "Class-based Cluster".
Refs of TKG document: https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/2.4/using-tkg/config-ref.html#kubernetes-tuning-classbased-clusters-only-5

File: templates/resources/tanzu_kubernetes_cluster.md.tmpl [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+```
+locals {
+  tkgm_v110_cluster_variables = { ... }
+  tkgm_v110_nodepool_a_overrides = { ... }
+  tkgs_cluster_variables = { ... }
+  tkgs_nodepool_a_overrides = { ... }
+}
+```
+
+# TKGM
```
[the remark this answers is not in the document]
Comment (Axpz) [reply templates/resources/tanzu_kubernetes_cluster.md.tmpl:22]: Should we breakdown into TKGm Azure/AWS/Vshpere, as well as the examples of xxx.tf? cc @ramya-bangera @willisren

[the remark this answers is not in the document]
Comment (vrabbi) [reply templates/resources/tanzu_kubernetes_cluster.md.tmpl:22]: uTKG support is ONLY for vSphere in the terraform provider and therefore it i believe is not needed to be called out. especially as Azure and AWS are deprecated in TKG itself.
the naming is being changed in the docs to remove reference to uTKG and use the names as has already been changed in the code itself in the other PRs.
@GilTeraSky please make sure to align the naming here as well in the docs.

File: examples/resources/tanzukubernetescluster/tkgs_vsphere_cluster_variables.tf [hunk 3]
```diff
@@ -0,0 +1,75 @@
+locals {
+  tkgs_cluster_variables = {
+    "TKR_DATA" : {
```
[the remark this answers is not in the document]
Comment (vrabbi) [reply examples/resources/tanzukubernetescluster/tkgs_vsphere_cluster_variables.tf:3]: this is part of the clusterclass returned from TMC API for TKGs but not for TKGm. the example is based on the defaults. we can remove it if needed

**Review by Axpz — APPROVED** [review]

**Review by vrabbi — APPROVED** [review]
LGTM

**Pull request merged**